instrumentation/methods/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/methods/MethodInstrumentationModule.java

```diff
@@ -27,27 +27,36 @@ public class MethodInstrumentationModule extends InstrumentationModule {
 
   public MethodInstrumentationModule() {
     super("methods");
+    typeInstrumentations = createInstrumentations();
+  }
+
+  private static List<TypeInstrumentation> createInstrumentations() {
+    DeclarativeConfigProperties methods =
+        AgentInstrumentationConfig.get().getDeclarativeConfig("methods");
+    List<TypeInstrumentation> list =
+        methods != null ? MethodsConfig.parseDeclarativeConfig(methods) : parseConfigProperties();
+    if (list.isEmpty()) {
+      return Collections.singletonList(new MethodInstrumentation(null, Collections.emptyMap()));
```

Comment:
I think there should be a comment here elaborating whey we are doing it.
Also I thought maybe it would be nicer if instead of `Map<String, SpanKind>` we'd use `Map<SpanKind, Collection<String>>`. Tried it out in
https://github.com/zeitlinger/opentelemetry-java-instrumentation/pull/6 wdyt?

Reply:
great idea - thanks!